client/src/components/users/EditUserProfileWrapper.tsx

```diff
@@ -0,0 +1,106 @@
+import React, {Component} from "react";
+import {match} from "react-router";
+import {Query} from "react-apollo";
+import gql from "graphql-tag";
+import {Header, Loader, Message} from "semantic-ui-react";
+import {User} from "../../actions";
+import UserProfile from "./UserProfile";
+import {AppState} from "../../reducers/reducers";
+import {connect} from "react-redux";
+
+interface EditUserProps {
+    match: match & EditUserParams;
+    user: User | null;
+}
+
+interface EditUserParams {
+    params: { userId?: string };
+}
+
+
+const USER_QUERY = gql`
+    query users($uuid:String!) {
+        users(search:{uuid:$uuid}) {
+            uuid
+            name
+            email
+            phone
+            slackUsername
+            haveID
+            admin
+        }
+    }
+`;
+
+class EditUserProfileWrapper extends Component<EditUserProps, {}> {
+    constructor(props: EditUserProps) {
+        super(props);
+    }
+
+
+    public render() {
+        const loader = <Loader active inline="centered" content="Loading profile information..."/>;
+
+        let userId: string | undefined = this.props.match.params.userId;
+        const title = (userId === undefined) ? "My Profile" : "Edit User";
+
+        const header = <Header size="huge">{title}</Header>;
+        console.log("user", userId === undefined, this.props.user);
+        if (userId === undefined && !this.props.user) {
+            return (
+                <div>
+                    {header}
+                    {loader}
+                </div>
+            );
+        }
+
+        if (!userId && this.props.user) {
+            userId = this.props.user.uuid;
+        }
+
+
+        return (
+            <div>
+                <Query
+                    query={USER_QUERY}
+                    variables={
+                        {
+                            uuid: userId
+                        }
+                    }
+                    fetchPolicy="no-cache"
+                >
+                    {
+                        ({loading, error, data}: any) => {
+                            if (loading) {
+                                return loader;
+                            } else if (error) {
+                                return <Message error visible={true}
+                                                header="Glitch in the matrix"
+                                                content={error.message}
+                                />;
+                            }
+                            {
+                                console.log(data);
+                            }
+                            return <UserProfile preloadUser={data.users[0]}/>;
+
+                        }
+                    }
+                </Query>
+            </div>
+        );
+    }
+}
+
+function mapStateToProps(state: AppState) {
+    return {
+        user: state.user
+    };
+}
+
+export default connect(
+    mapStateToProps
```

Comment:
so what happens when a random user goes to another person's id page? I don't know how I feel about even admin's getting access to another user's edit profile

Reply:
re: random user I see the private route logic below, admin concern stands

Reply:
I guess it doesn't hurt

Reply:
It shouldn't work because normal users can only access /user/me, while the /user/:I'd route is private to admins

Reply:
Oh, and I think it's standard for admins to be able to see/edit user information.